Add Feature: a ramses_cc.add_command action to add commands outside the config flow

IMMRMKW opened the pull request:
There are Ramses commands that allow a remote to set the ventilation unit to any percentage between 44% and 89.5%, with a defined resolution. For the Orcon ventilation unit, I identified the relevant command to be `31E0`.

While it's straightforward to calculate the required command for a specific percentage using a Python script and transmit it via the remote, the current implementation of ramses_cc does not support this functionality directly.

Although I believe the ideal solution would be to send arbitrary command strings to the remote in real time, I haven't implemented this yet—mainly because I haven't found a way to do this yet. Instead, I devised a workaround by introducing a new action in ramses_cc called `add_command`. Here's how it can be used:

```yaml
service: ramses.add_command
target:
  entity_id: remote.29_162275
data:
  command: setpoint  
  packet_string: " I --- 29:162275 32:146231 --:------ 31E0 008 000046000100AA00"
```

Once registered, the command can be triggered as follows:

```yaml
action: ramses_cc.send_command
target:
  entity_id: remote.29_162275
data:
  command: setpoint
  num_repeats: 3     # optional (default 3)
  delay_secs: 0.05   # optional (default 0.05)
```

I made the following changes in the code for this:

- Added SVC_ADD_COMMAND and SCH_ADD_COMMAND schemas in schemas.py
- Implemented async_add_command method in remote.py for direct command injection
- Removed unused imports and applied consistent code formatting

Note: All ramses_cc actions are also exposed under the remote namespace (e.g., `remote.learn_command`). However, the new `add_command` action is not yet available under that alias.

silverailscolo commented:
Thanks. To understand the approach:
The new action is named 'add_command', so it adds it to.. "the system"?

Should a user run that action (can be automated) after every reboot of HA? And where can the user see what has been added, where is it stored?

> Note: All ramses_cc actions are also exposed under the remote namespace (e.g., remote.learn_command).

That only applies for some, because those override built-in hvac actions (annotated in the _cc service.py code).

IMMRMKW commented:
I currently copied the approach used in `learn_command`, except this action has a field where the user - or an automation - can freely add a string, which is supposed to be a command. Currently, it is not possible to see which commands have been added, nor are they saved in a way that they are still available after a reboot. (Is this also the case for `learn_command`?)

My intentions were for new commands to be available for my fake remote, but indeed, the action is now `ramses_cc.add_command`, so I guess they are available within the integration. The idea is indeed that commands can be added in automated fashion, so that the user does not have to generate big lists of commands himself. The ideal solution would be to just generate commands programmatically, such that these do not have to be saved somewhere, and can be generated real time, but this is a first step towards something like that.

silverailscolo commented on custom_components/ramses_cc/remote.py:
```diff
@@ -142,6 +154,11 @@ async def async_learn_command(
             await self.async_delete_command(command)
 
+        # If packet_string is provided, add it directly without listening
+        if packet_string:
```
Not a problem, but a risk, think about adding a warning or extra confirmation step, making users aware they can upset their hardware

silverailscolo commented on tests/tests_old/test_services.py:
```diff
@@ -81,7 +81,7 @@
 
 NUM_DEVS_BEFORE = 3  # HGI, faked THM, faked REM
 NUM_DEVS_AFTER = 15  # proxy for success of cast_packets_to_rf()
-NUM_SVCS_AFTER = 10  # proxy for success
```
This is to be expected, since some tests compare to a snapshot to check remotes, and some checks count something and compare before/after

silverailscolo commented on tests/tests_old/test_services.py:
```diff
@@ -81,7 +81,7 @@
 NUM_DEVS_BEFORE = 3  # HGI, faked THM, faked REM
 NUM_DEVS_AFTER = 15  # proxy for success of cast_packets_to_rf()
-NUM_SVCS_AFTER = 10  # proxy for success
+NUM_SVCS_AFTER = 11  # proxy for success (includes add_command service)
```
includes a lot more, so please take this (note) out in a follow-up PR. OK for now

silverailscolo reviewed (approved):
Interesting new feature

IMMRMKW commented:
The changes in `learn_command` should not have been pushed, this was my original way of implementing the feature but I foresaw that it was a bit unhandy so I implemented it in a new function later. I cleaned up the code.
